src/errors.ts

```diff
@@ -1,40 +1,57 @@
 export class MalanError extends Error {
-  detail: string;
+  code: ErrorBody["code"];
+  detail: ErrorBody["detail"];
+  token_expired?: ErrorBody["token_expired"];
+  errors?: ErrorBody["errors"];
 
-  constructor(data: { detail: string; message: string }) {
+  constructor(data: ErrorBody) {
     super(data.message);
+    this.code = data.code;
     this.detail = data.detail;
+    this.token_expired = data.token_expired;
+    this.errors = data.errors;
   }
 }
 
+type DataBody<T = unknown> = {
+  data: T;
+};
+
+type ErrorBody = {
+  ok: false;
+  code: 400 | 401 | 403 | 404 | 422 | 423 | 429 | 461 | 462 | 500;
+  detail:
+    | "Bad Request"
+    | "Unauthorized"
+    | "Forbidden"
+    | "Not Found"
+    | "Unprocessable Entity"
+    | "Locked"
+    | "Too Many Requests"
+    | "Terms of Service Required"
+    | "Privacy Policy Required"
+    | "Internal Server Error";
+  message: string;
+  token_expired?: true;
```

Comment:
it looks like this can never be `false`

Reply:
ah yes, that's true, in the situations where the token is not expired, that key is never added.  From the client perspective does that feel like bad design?  I'm not really a fan of "sometimes there, sometimes not" elements.  I'm totally happy to think about how to improve it

Reply:
in this case i think `false` and `undefined` have the same meaning so it's probably okay